tests/test_queries.py

```diff
@@ -45,3 +46,21 @@ def test_filter_active_nodes_by_service_type(self):
             "openvpn"
         ).all()
         self.assertEqual([openvpn_node], openvpn_nodes)
+
+    def test_filter_nodes_in_bounty_programme(self):
+        dummy_node = Node("node1", "dummy")
+        dummy_node.mark_activity()
+        db.session.add(dummy_node)
+
+        dummy_node_2 = Node("node2", "dummy")
+        dummy_node_2.mark_activity()
+        db.session.add(dummy_node_2)
+
+        nodes = filter_active_nodes()
+        filtered = filter_nodes_in_bounty_programme(nodes).all()
+        self.assertEqual([], filtered)
+
+        ir = IdentityRegistration("node1", "test")
+        db.session.add(ir)
+        filtered = filter_nodes_in_bounty_programme(nodes).all()
+        self.assertEqual([dummy_node], filtered)
```

Comment:
I know it's a lot of duplication, but I would split this test-case into 2 - one with no identity registration, another one with identity registration. Because it's basically 2 separate cases, merged into one to avoid duplication, which is fine in tests.

Reply:
tests split